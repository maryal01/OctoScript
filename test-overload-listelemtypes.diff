1c1
< max of int list: 94; max of float list 3942650780012867434398622484716007093395947245117656288021316503468165889135026199245118992953728901437387245189226959438603365058750388395836362059587584477528268128697771159275255690961123755877584289203361977776530662635471208710144.00
---
> max of int list: 94; max of float list 83.82
