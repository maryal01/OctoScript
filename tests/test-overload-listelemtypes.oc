fn overload max (list<int> li) -> int {
    int max = li.list_get(0);
    int i = 1;
    while (i < li.list_length()) {
        int v = li.list_get(i);
        if (v > max){
            max = v; 
        }
        i = i + 1;
    }
    return max;
}

fn overload max (list<float> lf) -> float {
    float max = lf.list_get(0);
    int i = 1;
    while (i < lf.list_length()) {
        float v = lf.list_get(i);
        if (v > max){
            max = v; 
        }
        i = i + 1;
    }
    return max;
}

list<int> ilist = [13, 79, 94, 20, 39, 82, 23];
list<float> flist = [83.82, 58.33, 78.19, 4.48, 56.59, 79.70, 18.88];

print("max of int list: %d; max of float list %.2f\n", max(ilist), max(flist));
/* prints: max of int list: 94; max of float list 83.82 */