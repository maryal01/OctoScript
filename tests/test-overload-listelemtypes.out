max of int list: 94; max of float list 83.82
